### PR TITLE
REF/TST: Add more pytest idiom to indexing/multiindex/test_getitem.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ n/a] xref #24040
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ n/a] whatsnew entry

in this pass:
- follow-on from #24409

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_getitem.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import numpy as np
 import pytest
 
-from pandas.compat import StringIO, lrange, range, u, zip
+from pandas.compat import range, u, zip
 
 import pandas as pd
 from pandas import DataFrame, Index, MultiIndex, Series
-import pandas.core.common as com
```
Comment (jschendel) [inline pandas/tests/indexing/multiindex/test_getitem.py]: our linting checks are configured to enforce that anything used from `pandas.core.common` is imported in this manner

File: pandas/tests/indexing/multiindex/test_getitem.py [hunk 2]
```diff
@@ -206,116 +206,114 @@ def test_series_getitem_corner_generator(
...
+    df = multiindex_dataframe_random_data.T
+    expected = df.reindex(columns=df.columns[expected_slice])
     expected.columns = expected.columns.droplevel(0)
+    result = indexer(df)
     tm.assert_frame_equal(result, expected)
-    tm.assert_frame_equal(result, result2)
 
 
-def test_getitem_int(multiindex_dataframe_random_data):
+@pytest.fixture
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_getitem.py]: ideally put the fixtures at the top

File: pandas/tests/indexing/multiindex/test_getitem.py [hunk 3]
```diff
@@ -206,116 +206,114 @@ def test_series_getitem_corner_generator(
...
     with pytest.raises(com.SettingWithCopyError, match=msg):
         df['foo']['one'] = 2
 
+
+def test_frame_setitem_copy_no_write(multiindex_dataframe_random_data):
+    frame = multiindex_dataframe_random_data.T
+    expected = frame
+    df = frame.copy()
+
     try:
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_getitem.py]: don't do things like this; use the context manager like above

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/multiindex/test_getitem.py (hunk 3 above)
Comment (simonjayhawkins) [reply pandas/tests/indexing/multiindex/test_getitem.py]: i've changed as requested. though not convinced this is the right thing to do in situations like this. the previous test tested that the chained assignment raised the correct exception and this test is to check that the dataframe values are unchanged. having a context manager in the test adds unnecessary/unseen setup and teardown code that could potentially interfere with the tested functionality.

**Review by jreback — APPROVED** [review]

File: pandas/tests/indexing/multiindex/test_getitem.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/indexing/multiindex/test_getitem.py]: yeah this must have been changed a while back, the context manager is the correct idiom

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**